src/main/java/cms/gov/madie/measure/services/GroupService.java

```diff
@@ -308,9 +309,64 @@ private TestCaseStratificationValue updateTestCaseStratification(
               .build();
     }
     testCaseStrata.setName(strataName);
+    handlePopulationChange(testCaseStrata, testCaseGroup);
+
     return testCaseStrata;
   }
 
+  private void handlePopulationChange(
+      TestCaseStratificationValue testCaseStrata, TestCaseGroupPopulation testCaseGroup) {
+    List<TestCasePopulationValue> testCasePopulationValues = testCaseStrata.getPopulationValues();
+    List<TestCasePopulationValue> testCasePopulationValuesFromGroup =
+        testCaseGroup.getPopulationValues();
+
+    if (!CollectionUtils.isEmpty(testCasePopulationValuesFromGroup)) {
+      if (!CollectionUtils.isEmpty(testCasePopulationValues)) {
+        // when there is added group population
+        if (testCasePopulationValuesFromGroup.size() > testCasePopulationValues.size()) {
```

Comment:
what happens when a group is updated where there are multiple optional populations and one is swapped? For example, in the same group update a denominator exclusion is removed but a numerator exclusion is added?

Reply:
And can we add a unit test to cover that to make sure the logic works properly?

Reply:
@nmorasb This is done.